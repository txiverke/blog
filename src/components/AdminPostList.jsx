// @flow

import React from 'react'
import { connect } from 'react-redux'
import ReactMessages from 'react-messages'

import AdminPostItem from './AdminPostItem'
import AdminList from '../containers/AdminList'
import Loader from './Loader'
import { loadPostData } from '../actions/postActionCreators'

type Props = {
  dispatch: Function,
  posts: Data,
  handleClickCreate: Function,
}

class AdminPostList extends React.Component<Props> {
  props: Props
  
  componentDidMount() {
    const { posts, dispatch } = this.props

    if (posts.data.length === 0) {
      dispatch(loadPostData())
    }
  }

  createPost = (obj: Object) => {
    this.props.handleClickCreate(obj)
  }

  render() {
    const { data, completed, message, error } = this.props.posts

    if (completed) {
      return (
        <div className="app-content-grid"> 
          <ReactMessages 
            message={message} 
            error={error} 
            next={true} 
          />
          <AdminPostItem 
            label="Create Post" 
            handlePost={obj => this.createPost(obj)} 
          />
          <AdminList type="posts" data={data} />
        </div>
      )
    }

    return <Loader msg={message} />
  }
}

export const Unwrapped = AdminPostList
export default connect()(AdminPostList)