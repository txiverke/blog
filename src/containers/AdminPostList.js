// @flow

import { connect } from 'react-redux'

import AdminPostList from '../components/AdminPostList'
import { createPostData } from '../actions/postActionCreators'

const mapStateToProps = (state: Object) => ({ posts: state.posts })

const mapDispatchToProps = (dispatch: Function) => ({
  handleClickCreate: (obj: Object) => dispatch(createPostData(obj))
})

export default connect(mapStateToProps, mapDispatchToProps)(AdminPostList)