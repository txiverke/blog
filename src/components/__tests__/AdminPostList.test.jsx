// @flow

import React from 'react'
import { shallow } from 'enzyme'
import { Unwrapped as UnwrappedAdminPostList } from '../AdminPostList'

const properties = {
  dispatch: jest.fn(),
  posts: {
    completed: true,
    data: [],
    message: 'fake message',
    error: false
  },
  handleClickCreate: jest.fn()
}

describe('<AdminPostList />', () => {
  it('should render itself', () => {
    const component = shallow(<UnwrappedAdminPostList {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should render Loader if there is not data', () => {
    properties.posts.completed = false
    const component = shallow(<UnwrappedAdminPostList {...properties} />)
    expect(component.find('Loader').length).toEqual(1)  
  })

  it('should render the list if there is data', () => {
    properties.posts.completed = true
    const component = shallow(<UnwrappedAdminPostList {...properties} />)
    expect(component.find('Loader').length).toEqual(0)
    expect(component.find('.app-content-grid').length).toEqual(1)      
  })

})