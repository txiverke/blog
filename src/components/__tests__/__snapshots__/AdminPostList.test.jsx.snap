// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPostList /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPostList
    dispatch={[Function]}
    handleClickCreate={[Function]}
    posts={
        Object {
            "completed": true,
            "data": Array [],
            "error": false,
            "message": "fake message",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <M
          duration={2000}
          error={false}
          icon="heart"
          message="fake message"
          next={true}
/>,
        <AdminPostItem
          handlePost={[Function]}
          label="Create Post"
/>,
        <Connect(AdminList)
          data={Array []}
          type="posts"
/>,
      ],
      "className": "app-content-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "duration": 2000,
          "error": false,
          "icon": "heart",
          "message": "fake message",
          "next": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handlePost": [Function],
          "label": "Create Post",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [],
          "type": "posts",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <M
            duration={2000}
            error={false}
            icon="heart"
            message="fake message"
            next={true}
/>,
          <AdminPostItem
            handlePost={[Function]}
            label="Create Post"
/>,
          <Connect(AdminList)
            data={Array []}
            type="posts"
/>,
        ],
        "className": "app-content-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "duration": 2000,
            "error": false,
            "icon": "heart",
            "message": "fake message",
            "next": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handlePost": [Function],
            "label": "Create Post",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "type": "posts",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
